wpilibc/athena/src/Notifier.cpp

```diff
@@ -31,7 +31,10 @@ Notifier::Notifier(TimerEventHandler handler) {
  */
 Notifier::~Notifier() {
   int32_t status = 0;
-  cleanNotifier(m_notifier, &status);
+  // atomically set handle to 0, then clean
+  HalNotifierHandle handle = m_notifier;
+  m_notifier = 0;
+  cleanNotifier(handle, &status);
```

Comment:
This is not an atomic operation.  You need to instead do `handle = m_notifier.exchange(0)` and then check if handle is 0 to atomically set handle to 0.